assets/src/apps/authoring/components/PropertyEditor/schemas/lesson.ts

```diff
@@ -319,7 +319,7 @@ export const transformModelToSchema = (model: any) => {
       title: model.title,
       customCSS: model.customCss,
       enableHistory: model.custom.allowNavigation || model.custom.enableHistory || false,
-      displayRefreshWarningPopup: model.custom.displayRefreshWarningPopup || true,
```

Comment:
By default, `displayRefreshWarningPopup` property is true. When user was unchecking it, the condition here was getting set as 'false || true' and due to that the check box was getting checked.